sentry_openproject/plugin.py

```diff
@@ -0,0 +1,152 @@
+        widget=forms.TextInput(attrs={'class': 'span9'}),
+        help_text=_('Slug of the project on OpenProject'),
+    )
+    assignee_id = forms.IntegerField(
+        label=_('OpenProject Assignee ID'),
+        widget=forms.TextInput(attrs={'class': 'span9'}),
+        required=False, min_value=1,
+        help_text=_(
+            'ID of the OpenProject user who will be the default assignee'),
+    )
+
+    def clean(self):
+        config = self.cleaned_data
+        if not all(config.get(k) for k in (
+                'host', 'apikey', 'project_slug', 'assignee_id')):
+            raise forms.ValidationError('Missing required configuration value')
+        return config
+
+
+class OpenProjectNewIssueForm(forms.Form):
+    title = forms.CharField(
+        max_length=200, widget=forms.TextInput(attrs={'class': 'span9'}))
+    description = forms.CharField(
+        widget=forms.Textarea(attrs={'class': 'span9'}))
+
+
+class OpenProjectPlugin(IssuePlugin):
+    author = 'HBEE'
+    author_url = 'https://github.com/HBEE/sentry-openproject'
+    version = '0.1.0'
+    description = (
+        'Integrate OpenProject issue tracking by linking a user account '
+        'to a project.')
+    resource_links = [
+        ('Bug Tracker', 'https://github.com/HBEE/sentry-openproject/issues'),
+        ('Source', 'https://github.com/HBEE/sentry-openproject'),
+    ]
+
+    slug = 'openproject'
+    title = _('OpenProject')
+    conf_title = 'OpenProject'
+    conf_key = 'openproject'
+    project_conf_form = OpenProjectOptionsForm
+    new_issue_form = OpenProjectNewIssueForm
+
+    def is_configured(self, project, **kwargs):
+        return all(
+            (self.get_option(k, project) for k in (
+                'host', 'apikey', 'project_slug', 'assignee_id')))
+
+    def get_new_issue_title(self, **kwargs):
+        return 'Create OpenProject Task'
+
+    def get_initial_form_data(self, request, group, event, **kwargs):
+        return {
+            'title': 'Sentry:%s' % self._get_group_title(request, group, event),
+            'description': self._get_group_description(request, group, event),
+        }
+
+    def _get_group_description(self, request, group, event):
+        output = [
+            absolute_uri(group.get_absolute_url()),
+        ]
+        body = self._get_group_body(request, group, event)
+        if body:
+            output.extend([
+                '',
+                '<pre>',
+                body,
+                '</pre>',
+            ])
+        return '\n'.join(output)
+
+    def create_issue(self, group, form_data, **kwargs):
+        """Create an issue on OpenProject"""
+        host = self.get_option('host', group.project)
+        apikey = self.get_option('apikey', group.project)
+        project = self.get_option('project_slug', group.project)
+        assignee = self.get_option('assignee_id', group.project)
+
+        url = urlparse.urljoin(
+            host, '/api/v3/projects/%s/work_packages' % project)
```

Comment:
I see that '/api/v3' is used in line 137 could we have a separate var, because in future version of api can change. What you think, if it's worth?